### PR TITLE
(feat) O3-1890: Integrate annotation functionality via a drawing widget

### PR DESCRIPTION
**Pull request by jona42-ui**
This task PR  focuses on implementing the Innitial diagram annotation functionality 

The objective of this task is to integrate diagram annotation functionality into By incorporating a suitable frontend library, such as React Image Annotate, which will be leveraged. users will be able to annotate the loaded diagrams/images.
more tasks to improve may come in follow up separate PRs

### CODE REVIEW

**Comment by jona42-ui** [comment]
Screenshot from 2023-07-07 08-11-53

**Comment by jona42-ui** [comment]
KIndly seeking for your review

@denniskigen @jayasanka-sack @heshanandrews @ibacher

File: src/draw-page.scss [hunk 1]
```diff
@@ -0,0 +1 @@
+@import url("https://rsms.me/inter/inter.css");
```
Comment (denniskigen) [inline src/draw-page.scss:1]: We shouldn't need to use a custom font. We want to use the same font (IBM Plex Sans) across the application for consistency.

**Comment by denniskigen** [comment]
Not a huge fan of having empty files (the `resource` file and the `utils` file). Add them as the need arises. Also, using the `scss` prefix for your stylesheets instead of the `*.style.css` nomenclature is preferable. Find more guidance on coding conventions [here](https://o3-docs.vercel.app/docs/coding-conventions).

File: src/components/drawing-widget/drawing-widget.component.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  const [annotations, setAnnotations] = useState<Annotation[]>([]);
+
+  const handleAnnotationUpdate = (updatedAnnotations: Annotation[]) => {
+    setAnnotations(updatedAnnotations);
+  };
+
+  const handleSaveAnnotations = () => {
+    // Perform actions with the captured annotations
+    // eslint-disable-next-line no-console
+    console.log("Annotations:", annotations);
```
Comment (denniskigen) [inline src/components/drawing-widget/drawing-widget.component.tsx:37]: Please remove this `console.log`.

File: src/components/drawing-widget/drawing-widget.component.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+interface DrawingWidgetProps {
+  selectedImage: string;
+  taskDescription: string;
+  images: { src: string; name: string }[];
+  regionClsList?: string[];
+  enabledTools?: string[];
+}
+
+const DrawingWidget: React.FC<DrawingWidgetProps> = ({
+  selectedImage,
```
Comment (denniskigen) [inline src/components/drawing-widget/drawing-widget.component.tsx:22]: If we're not using these props, we might as well remove them.

File: src/draw-page.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+@import '~@openmrs/esm-styleguide/src/vars';
```
Comment (samuelmale) [inline src/draw-page.scss:1]: @jona42-ui are you using this file

File: src/components/drawing-widget/drawing-widget.component.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      <ReactImageAnnotate
+        labelImages
+        regionClsList={["Alpha", "Beta", "Charlie", "Delta"]}
+        regionTagList={["tag1", "tag2", "tag3"]}
+        images={[
+          {
+            src: "https://placekitten.com/408/287",
+            name: "Cat Physical Exam",
+            regions: [],
+          },
```
Comment (samuelmale) [inline src/components/drawing-widget/drawing-widget.component.tsx:24]: You probably don't want to persist this to source control but if so, I would add a TODO that bounds me to remove this later.

**Comment by jona42-ui** [comment]
Thanks @samuelmale @denniskigen am learning alot from you.

**Pull request closed without merging**